torchmetrics/classification/auroc.py

```diff
@@ -118,20 +119,30 @@ def __init__(
         max_fpr: Optional[float] = None,
         compute_on_step: bool = True,
         dist_sync_on_step: bool = False,
+        reduce_memory: bool = False,
```

Comment:
This is strictly always a reduction. For a very small dataset and a moderately large threshold count, it will actually increase memory usage. Maybe `fixed_memory` would be better?